### PR TITLE
Determine checkmate

### PR DESCRIPTION
**Pull request by adrianproctor**
Initial commit for checkmate? method

### CODE REVIEW

**Comment by adrianproctor** [comment]
*** Work in progress **** 
Todo: refactor to reduce ABC complexity; add additional test cases

**Comment by adrianproctor** [comment]
Ready for code review

File: spec/models/game_spec.rb [hunk 1]
```diff
@@ -6,6 +6,102 @@
...
+    it 'should return false - black rook can block diagonal attack from white bishop' do
+      FactoryGirl.create(:king, game: game2, row_coordinate: 3, column_coordinate: 0, color: 'black')
+      FactoryGirl.create(:king, game: game2, row_coordinate: 7, column_coordinate: 7, color: 'white')
+      FactoryGirl.create(:queen, game: game2, row_coordinate: 0, column_coordinate: 1, color: 'white')
+      FactoryGirl.create(:rook, game: game2, row_coordinate: 7, column_coordinate: 1, color: 'black')
+      FactoryGirl.create(:rook, game: game2, row_coordinate: 4, column_coordinate: 7, color: 'white')
+      FactoryGirl.create(:rook, game: game2, row_coordinate: 2, column_coordinate: 7, color: 'white')
+      FactoryGirl.create(:bishop, game: game2, row_coordinate: 7, column_coordinate: 4, color: 'white')
+      expect(game2.checkmate?).to eq false
+    end
```
Comment (mscoutermarsh) [inline spec/models/game_spec.rb]: great job with the tests :+1:

File: app/models/game.rb [hunk 2]
```diff
@@ -16,14 +16,35 @@ def in_check?
...
+    return true if attackers.count > 1
+
+    # determine if attacker can be captured
+    return false if king_attacker_can_be_captured?(attackers.first, friendly_list)
+
+    # determine if check can be blocked
+    return false if check_can_be_blocked?(king, attackers.first, friendly_list)
+
+    true
+  end
```
Comment (mscoutermarsh) [inline app/models/game.rb]: Reads very easily. I like how you extracted complexity into well named private methods. :+1:

**Comment by mscoutermarsh** [comment]
Looks great. Has merge conflicts, go ahead and merge whenever you're ready :)

**Pull request merged**